src/intermediate_structures.h

```diff
@@ -0,0 +1,85 @@
+/**
```

Comment:
Big fan of this change, makes the code **so** much nicer! 👍 

Only thing is perhaps adding a bit of documentation to the undocumented structs/arrays. I don't mind accepting this PR and then adding this in myself, but I think there should be some explanation why we have two `Coordinate` structs)

Reply:
(This was meant to be a comment for the whole file)

Reply:
Thanks, I've updated the documentation and tried to make the distinction between the coordinate types clearer.